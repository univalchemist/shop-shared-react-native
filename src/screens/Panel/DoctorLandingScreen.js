import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';
import { Box, Text, ScrollView, Button } from '@wrappers/components';
import { Image } from '@wrappers/components';
import { useIntl, useTheme } from '@wrappers/core/hooks';
import styled from 'styled-components/native';
import ServiceItem from './widgets/ServiceItem';
import SpecialistItem from './widgets/SpecialistItem';
import {
  DOCTOR_LISTING,
  PANEL_SEARCH,
  SCAN_QR_CODE,
  SPECIALITY_LISTING,
  UNIFY_SEARCH,
} from '@routes';
import { creditCardScanIcon } from '@images';
import {
  generalSurgeonImg,
  dermatologyImg,
  pediatricianImg,
  dentistImg,
  cardiologyImg,
  orthopedicsImg,
  urologyImg,
  counselingImg,
} from '@images/specialist';
import {
  axaPanelClinicImg,
  generalPractitionerImg,
  specialistImg,
} from '@images/doctor_landing';
import NonTouchableSearchBar from '@heal/src/components/NonTouchableSearchBar';
import { getSpecialities } from '@heal/src/store/actions';

const HeadingText = styled(Text)`
  font-size: 21px;
  line-height: 25px;
  ${({ theme }) => `
    color: ${theme.colors.gray[0]}
  `};
`;

const services = [
  {
    image: generalPractitionerImg,
    titleId: 'doctorLanding.service.generalPractitioner.title',
    descId: 'doctorLanding.service.generalPractitioner.desc',
    routeName: PANEL_SEARCH,
  },
  {
    image: specialistImg,
    titleId: 'doctorLanding.service.specialist.title',
    descId: 'doctorLanding.service.specialist.desc',
    routeName: DOCTOR_LISTING,
  },
];

const specialists = [
  {
    image: generalSurgeonImg,
    code: 'generalsurgery',
  },
  {
    image: dermatologyImg,
    code: 'dermatologyvener',
  },
  {
    image: pediatricianImg,
    code: 'paediatrics',
  },
  {
    image: dentistImg,
    code: 'dentist',
  },
  {
    image: cardiologyImg,
    code: 'cardiology',
  },
  {
    image: orthopedicsImg,
    code: 'orthopaedicstrau',
  },
  {
    image: urologyImg,
    code: 'urogynaecology',
  },
  {
    image: counselingImg,
    code: 'counsellingpsych',
  },
];

const DoctorLandingScreen = ({ navigation, firstName, getSpecialities }) => {
  const intl = useIntl();
  useEffect(() => {
    const fetchSpecialities = async () => {
      await getSpecialities();
    };
    fetchSpecialities();
  }, []);
  return (
    <Box flex={1} mt={24}>
      <ScrollView>
        <Box px={32} pb={24}>
          <Box
            flexDirection="row"
            justifyContent="space-between"
            alignItems="center"
          >
            <NonTouchableSearchBar
              onPress={() => navigation.navigate(UNIFY_SEARCH)}
              placeholder={intl.formatMessage({
                id: 'doctorSearch.placeholder',
              })}
            />
            <Box flex={0.2} alignItems="center">
              <TouchableOpacity
                onPress={() => navigation.navigate(SCAN_QR_CODE)}
              >
                <Image source={creditCardScanIcon} />
              </TouchableOpacity>
            </Box>
          </Box>
          <Box my={24}>
            <HeadingText>
              {intl.formatMessage(
                {
                  id: 'doctorLanding.panelSearchQuestion',
                },
                {
                  firstName: firstName,
                },
              )}
            </HeadingText>
          </Box>
          <Box py={12}>
            {services.map((service, index) => (
              <ServiceItem {...service} navigation={navigation} key={index} />
            ))}
          </Box>
          <Box my={24}>
            <HeadingText>
              {intl.formatMessage({
                id: 'doctorLanding.findASpecialist',
              })}
            </HeadingText>
          </Box>
          <Box
            flexWrap="wrap"
            flexDirection="row"
            justifyContent="space-between"
          >
            {specialists &&
              specialists.map((specialist, index) => (
                <Box flexBasis={0.47} key={index}>
                  <SpecialistItem
                    {...specialist}
                    onPress={() => {
                      navigation.navigate(DOCTOR_LISTING, {
                        code: specialist.code,
                      });
                    }}
                  />
                </Box>
              ))}
          </Box>
          <Box pt={16} pb={24}>
            <Button
              secondary
              title={intl.formatMessage({
                id: 'doctorLanding.viewAllSpecialities',
              })}
              onPress={() => navigation.navigate(SPECIALITY_LISTING)}
            />
          </Box>
        </Box>
      </ScrollView>
    </Box>
  );
};

DoctorLandingScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
  }),
};

const mapStateToProps = ({ user: { firstName } }) => ({
  firstName,
});

export default connect(mapStateToProps, { getSpecialities })(
  DoctorLandingScreen,
);
