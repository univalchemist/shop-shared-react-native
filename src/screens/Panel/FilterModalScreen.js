import React from 'react';
import { connect } from 'react-redux';
import { CustomMultiselectCheckBox } from '@wrappers/components/form';
import { useIntl } from '@wrappers/core/hooks';
import { updateFilter } from '@store/panel/actions';
import { SPECIALTY } from './utils/filter/FilterTypes';
import { categories, logAction } from '@store/analytics/trackingActions';

const toTitleCase = str => {
  return str.replace(/\w\S*/g, function(txt) {
    return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
  });
};

export const FilterModalScreen = ({
  selectedValues,
  updateFilter,
  specialty,
  navigation,
}) => {
  const intl = useIntl();
  const data = specialty;
  return (
    <>
      <CustomMultiselectCheckBox
        buttonLabel={intl.formatMessage({ id: 'showResults' })}
        clearAllButtonLabel={intl.formatMessage({ id: 'clearAll' })}
        clearAllButtonTrackingActionParams={{
          category: categories.PANEL_CLINIC_SEARCH,
          action: 'Clear all filters',
        }}
        data={data}
        selectedValues={selectedValues}
        onSubmit={values => {
          logAction({
            category: categories.PANEL_CLINIC_SEARCH,
            action: `Filter search by ${toTitleCase(values.join(', '))}`,
          });
          updateFilter({ type: SPECIALTY, values });
          navigation.goBack();
        }}
        onChange={() => {}}
      />
    </>
  );
};

const mapStateToProps = ({ panel: { filters, specialty } }) => {
  return {
    selectedValues: filters.getFilterByType(SPECIALTY).values || [],
    specialty: specialty.map(current => ({
      label: toTitleCase(current),
      value: current,
    })),
  };
};

const enhance = connect(mapStateToProps, { updateFilter });

export default enhance(FilterModalScreen);
