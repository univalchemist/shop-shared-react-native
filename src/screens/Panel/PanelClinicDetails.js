import { Box, Text, ListItem, Image } from '@wrappers/components';
import { useIntl, useTheme } from '@wrappers/core/hooks';
import React from 'react';
import { ScrollView, Linking } from 'react-native';
import {
  phoneIcon,
  directionIcon,
  chevronRightArrow,
  warningIcon,
} from '@images';
import showDirection from './utils/ShowDirection';
import { categories, logAction } from '@store/analytics/trackingActions';
import styled from 'styled-components/native';
import FeatureToggle from '@config/FeatureToggle';

const TerminatedContainer = styled(Box)`
  ${({ theme }) => `
  font-size: ${theme.fontSizes[2]}
  background-color: ${theme.terminatedLabel.backgroundColor}
  padding: 16px
  border-radius: 4px
  `}
`;

const TerminatedText = styled(Text)`
  ${({ theme }) => `
  color: ${theme.colors.gray[0]}
  padding: 0 16px 0 11px
  `}
`;

const ImageBox = styled(Box)`
  padding-top: 3px;
`;

const PanelClinicDetails = ({ route }) => {
  const params = route?.params || {};
  const selectedClinic = params?.selectedClinic;
  const intl = useIntl();
  const theme = useTheme();

  const headerTextStyle = {
    color: theme.colors.gray[3],
  };

  const tryOpeningLink = url => {
    if (url) {
      Linking.canOpenURL(url).then(supported => {
        if (supported) {
          return Linking.openURL(url);
        }
        alert(`Link: ${url} Could not be opened`);
      });
    }
  };

  const isValueDefined = value => {
    return value?.length > 0;
  };

  const addBreakLine = (consultationIndex, selectedClinic) => {
    return consultationIndex !==
      selectedClinic.consultationTimings.length - 1 ||
      isValueDefined(selectedClinic.peakHour) ? (
      <Text />
    ) : null;
  };

  const logCallAction = () => {
    logAction({
      category: categories.PANEL_CLINIC_DETAILS,
      action: 'Call clinic from clinic details',
    });
  };

  return (
    <Box bg="gray.7" flex={1}>
      <ScrollView>
        <Box flex={1}>
          {FeatureToggle.TERMINATED_LABEL_FOR_CLINIC.on &&
            !!selectedClinic.terminationDate && (
              <Box px={32} py={24} flex={1}>
                <TerminatedContainer flexDirection={'row'}>
                  <ImageBox>
                    <Image source={warningIcon} />
                  </ImageBox>
                  <TerminatedText>
                    {intl.formatMessage({ id: 'panelSearch.terminationText' })}
                  </TerminatedText>
                </TerminatedContainer>
              </Box>
            )}
          {isValueDefined(selectedClinic.contactNumber1) && (
            <ListItem
              leftIcon={
                <Image source={phoneIcon} accessibilityLabel="Phone Icon" />
              }
              rightIcon={
                <Image
                  source={chevronRightArrow}
                  accessibilityLabel="Right Arrow Icon"
                />
              }
              onPress={() => {
                logCallAction();
                tryOpeningLink(`tel:${selectedClinic.contactNumber1}`);
              }}
            >
              <Text>{selectedClinic.contactNumber1}</Text>
            </ListItem>
          )}

          {isValueDefined(selectedClinic.contactNumber2) && (
            <ListItem
              leftIcon={
                <Image source={phoneIcon} accessibilityLabel="Phone Icon" />
              }
              rightIcon={
                <Image
                  source={chevronRightArrow}
                  accessibilityLabel="Right Arrow Icon"
                />
              }
              onPress={() => {
                logCallAction();
                tryOpeningLink(`tel:${selectedClinic.contactNumber2}`);
              }}
            >
              <Text>{selectedClinic.contactNumber2}</Text>
            </ListItem>
          )}

          {isValueDefined(selectedClinic.contactNumber3) && (
            <ListItem
              leftIcon={
                <Image source={phoneIcon} accessibilityLabel="Phone Icon" />
              }
              rightIcon={
                <Image
                  source={chevronRightArrow}
                  accessibilityLabel="Right Arrow Icon"
                />
              }
              onPress={() => {
                logCallAction();
                tryOpeningLink(`tel:${selectedClinic.contactNumber3}`);
              }}
            >
              <Text>{selectedClinic.contactNumber3}</Text>
            </ListItem>
          )}

          {isValueDefined(selectedClinic.address) && (
            <ListItem
              leftIcon={
                <Image
                  source={directionIcon}
                  accessibilityLabel="Direction Icon"
                />
              }
              rightIcon={
                <Image
                  source={chevronRightArrow}
                  accessibilityLabel="Right Arrow Icon"
                />
              }
              onPress={() => {
                logAction({
                  category: categories.PANEL_CLINIC_DETAILS,
                  action: 'Directions from clinic details',
                });

                showDirection(selectedClinic);
              }}
            >
              <Text>{selectedClinic.address}</Text>
            </ListItem>
          )}

          {isValueDefined(selectedClinic.specialty) && (
            <ListItem>
              <Text style={headerTextStyle}>
                {intl.formatMessage({ id: `panelSearch.clinicType` })}
              </Text>
              <Text style={{ textTransform: 'capitalize' }}>
                {selectedClinic.specialty}
              </Text>
            </ListItem>
          )}

          {isValueDefined(selectedClinic.consultationTimings) && (
            <ListItem>
              {selectedClinic.consultationTimings.map(
                (consultation, consultationIndex) =>
                  Object.keys(consultation).map((header, headerIndex) => (
                    <Box key={`${consultationIndex}_${headerIndex}`}>
                      <Text style={headerTextStyle}>{header}</Text>
                      {consultation[header].map((consultationTime, index) => (
                        <Text key={index}>{consultationTime}</Text>
                      ))}
                      {addBreakLine(consultationIndex, selectedClinic)}
                    </Box>
                  )),
              )}
              {isValueDefined(selectedClinic.peakHour) && (
                <Box>
                  <Text style={headerTextStyle}>
                    {intl.formatMessage({ id: `panelSearch.peakHours` })}
                  </Text>
                  <Text>{selectedClinic.peakHour}</Text>
                </Box>
              )}
            </ListItem>
          )}

          {isValueDefined(selectedClinic.language) && (
            <ListItem>
              <Text style={headerTextStyle}>
                {intl.formatMessage({ id: `panelSearch.languagesSpoken` })}
              </Text>
              <Text>{selectedClinic.language}</Text>
            </ListItem>
          )}
        </Box>
      </ScrollView>
    </Box>
  );
};

export default PanelClinicDetails;
