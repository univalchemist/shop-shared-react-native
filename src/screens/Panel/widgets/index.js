export { default as MultipleClinicsListView } from './MultipleClinicsListView';
export { default as DoctorLandingHeader } from './DoctorLandingHeader';
export { default as ServiceItem } from './ServiceItem';
export { default as TerminatedLabel } from './TerminatedLabel';
export { default as PanelListView } from './PanelListView';
