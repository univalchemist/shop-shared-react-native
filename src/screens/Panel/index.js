import PanelSearchScreen from './PanelSearchScreen';
import PanelClinicDetails from './PanelClinicDetails';
import FilterModalScreen from './FilterModalScreen';
import DoctorLandingScreen from './DoctorLandingScreen';

export {
  PanelSearchScreen,
  PanelClinicDetails,
  FilterModalScreen,
  DoctorLandingScreen,
};

export {
  MultipleClinicsListView,
  DoctorLandingHeader,
  ServiceItem,
  TerminatedLabel,
  PanelListView,
} from './widgets';
