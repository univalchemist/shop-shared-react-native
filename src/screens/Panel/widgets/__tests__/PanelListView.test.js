import React from 'react';
import { FlatList, TouchableHighlight } from 'react-native';
import { renderForTest } from '@testUtils';
import { Image, ErrorPanel } from '@wrappers/components';
import { fireEvent } from 'react-native-testing-library';
import { PANEL_CLINIC_DETAILS } from '@routes';
import { TerminatedLabel, PanelListView } from '@screens/Panel';

describe('PanelListView', () => {
  const clinics = [
    {
      id: 1,
      name: 'Dr abc',
      address: 'CommonWealth MRT',
      contactNumber1: '2223 2231',
      specialty: 'General Practice',
    },
    {
      id: 2,
      name: 'Dr xyz',
      address: 'City Hall MRT',
      contactNumber1: '2223 2232',
      contactNumber2: '2223 2233',
      contactNumber3: '2223 2234',
      specialty: 'General Surgery',
    },
  ];
  it('should match snapshot', () => {
    const panelListView = renderForTest(
      <PanelListView clinics={clinics} />,
    ).toJSON();
    expect(panelListView).toMatchSnapshot();
  });

  it('should show the list view', () => {
    const panelListView = renderForTest(<PanelListView clinics={clinics} />);

    expect(panelListView.queryAllByType(FlatList).length).toBe(1);
    expect(panelListView.queryAllByType(Image).length).not.toBeNull();
    expect(panelListView.queryByText(clinics[0].name)).not.toBeNull();
    expect(panelListView.queryByText(clinics[0].address)).not.toBeNull();
    expect(panelListView.queryByText(clinics[0].specialty)).not.toBeNull();
    expect(panelListView.queryByText(clinics[0].contactNumber1)).not.toBeNull();
    expect(panelListView.queryByText(clinics[1].name)).not.toBeNull();
    expect(panelListView.queryByText(clinics[1].address)).not.toBeNull();
    expect(panelListView.queryByText(clinics[1].specialty)).not.toBeNull();
    expect(
      panelListView.queryByText(
        `${clinics[1].contactNumber1}, ${clinics[1].contactNumber2}, ${clinics[1].contactNumber3}`,
      ),
    ).not.toBeNull();
  });

  it('should show the number of clinics listed', () => {
    const panelListView = renderForTest(<PanelListView clinics={clinics} />);
    const clinicsNumber = clinics.length;
    const clinicsNumberText = `${clinicsNumber} Clinics / Doctors found`;

    expect(panelListView.queryByText(clinicsNumberText)).not.toBeNull();
  });

  it('should show the detailed clinic panel when the clinic is clicked on', async () => {
    const navigation = { navigate: jest.fn() };
    const { queryAllByType } = renderForTest(
      <PanelListView clinics={clinics} navigation={navigation} />,
    );

    fireEvent.press(queryAllByType(TouchableHighlight)[1]);

    expect(navigation.navigate).toHaveBeenCalledTimes(1);
    expect(navigation.navigate).toHaveBeenCalledWith(PANEL_CLINIC_DETAILS, {
      selectedClinic: clinics[1],
    });
  });

  it('should show error panel with appropriate message if zero clinic found', () => {
    const panelListView = renderForTest(<PanelListView clinics={[]} />);

    expect(panelListView.queryAllByType(ErrorPanel).length).toBe(1);
  });

  it('should show error panel with appropriate message if fetching clinic failed', () => {
    const panelListView = renderForTest(<PanelListView clinics={null} />);

    expect(panelListView.queryAllByType(ErrorPanel).length).toBe(1);
  });
  it('should show - text if value is empty or unavailable', () => {
    const clinics = [
      {
        name: 'Dr abc',
        address: '',
        specialty: 'General Practice',
      },
      {
        name: 'Dr xyz',
        address: 'City Hall MRT',
        contactNumber1: '',
      },
      {
        name: 'Dr xyz',
        contactNumber1: '2223 2233',
        specialty: 'General Surgery',
      },
      {
        name: 'Dr xyza',
        contactNumber1: '',
        contactNumber2: '',
        contactNumber3: '',
        specialty: 'General Surgery',
        address: 'City Hall MRT',
      },
      {
        name: 'Dr xyza',
        contactNumber1: ' ',
        contactNumber2: ' ',
        contactNumber3: ' ',
        specialty: 'General Surgery',
        address: 'City Hall MRT',
      },
    ];
    const panelListView = renderForTest(<PanelListView clinics={clinics} />);

    expect(panelListView.queryAllByText('-').length).toBe(7);
  });

  describe('terminated label', () => {
    beforeEach(() => {
      const FeatureToggle = require('@config/FeatureToggle').default;
      FeatureToggle.TERMINATED_LABEL_FOR_CLINIC.turnOn();
    });
    it('should show clinic with termination text when terminationDate is not null', () => {
      const clinics = [
        {
          name: 'Dr abc',
          address: '',
          specialty: 'General Practice',
          terminationDate: '30Sep2019',
        },
      ];
      const panelListView = renderForTest(<PanelListView clinics={clinics} />);

      expect(panelListView.queryAllByType(TerminatedLabel).length).toBe(1);
    });

    it('should not show clinic with termination text when terminationDate is null', () => {
      const clinics = [
        {
          name: 'Dr abc',
          address: '',
          specialty: 'General Practice',
          terminationDate: null,
        },
      ];
      const panelListView = renderForTest(<PanelListView clinics={clinics} />);

      expect(panelListView.queryAllByType(TerminatedLabel).length).toBe(0);
    });
  });
});
