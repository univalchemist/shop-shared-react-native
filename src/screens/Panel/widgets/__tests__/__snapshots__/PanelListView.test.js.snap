// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelListView should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {},
                            Object {
                              "backgroundColor": "#f9f9f9",
                            },
                          ]
                        }
                      >
                        <RCTScrollView
                          ListHeaderComponent={
                            <ForwardRef
                              backgroundColor="gray.7"
                              pb={2}
                              pt={3}
                              px={4}
                            >
                              <SectionHeadingText
                                accessibilityLabel="Number of Clinics Found"
                              >
                                2 Clinics / Doctors found
                              </SectionHeadingText>
                            </ForwardRef>
                          }
                          data={
                            Array [
                              Object {
                                "address": "CommonWealth MRT",
                                "contactNumber1": "2223 2231",
                                "id": 1,
                                "name": "Dr abc",
                                "specialty": "General Practice",
                              },
                              Object {
                                "address": "City Hall MRT",
                                "contactNumber1": "2223 2232",
                                "contactNumber2": "2223 2233",
                                "contactNumber3": "2223 2234",
                                "id": 2,
                                "name": "Dr xyz",
                                "specialty": "General Surgery",
                              },
                            ]
                          }
                          disableVirtualization={false}
                          getItem={[Function]}
                          getItemCount={[Function]}
                          horizontal={false}
                          initialNumToRender={100}
                          keyExtractor={[Function]}
                          maxToRenderPerBatch={10}
                          numColumns={1}
                          onContentSizeChange={[Function]}
                          onEndReachedThreshold={0.5}
                          onLayout={[Function]}
                          onMomentumScrollEnd={[Function]}
                          onScroll={[Function]}
                          onScrollBeginDrag={[Function]}
                          onScrollEndDrag={[Function]}
                          removeClippedSubviews={false}
                          renderItem={[Function]}
                          scrollEventThrottle={50}
                          stickyHeaderIndices={
                            Array [
                              0,
                            ]
                          }
                          updateCellsBatchingPeriod={50}
                          viewabilityConfigCallbackPairs={Array []}
                          windowSize={21}
                        >
                          <View>
                            <View
                              onLayout={[Function]}
                            >
                              <View
                                backgroundColor="gray.7"
                                pb={2}
                                pt={3}
                                px={4}
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "#f9f9f9",
                                      "paddingBottom": 8,
                                      "paddingLeft": 32,
                                      "paddingRight": 32,
                                      "paddingTop": 16,
                                    },
                                  ]
                                }
                              >
                                <Text
                                  accessibilityLabel="Number of Clinics Found"
                                  color="gray.0"
                                  fontFamily="default"
                                  fontSize={16}
                                  fontWeight="bold"
                                  lineHeight={22}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#212121",
                                        "fontFamily": "Arial",
                                        "fontSize": 16,
                                        "fontWeight": "700",
                                        "lineHeight": 22,
                                      },
                                    ]
                                  }
                                >
                                  2 Clinics / Doctors found
                                </Text>
                              </View>
                            </View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                accessibilityLabel="Dr abc"
                                accessible={true}
                                focusable={true}
                                isTVSelectable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={null}
                              >
                                <View
                                  alignItems="center"
                                  flexDirection="row"
                                  px={4}
                                  py={24}
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "paddingBottom": 24,
                                        "paddingLeft": 32,
                                        "paddingRight": 32,
                                        "paddingTop": 24,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    flex={1}
                                    style={
                                      Array [
                                        Object {
                                          "flexBasis": 0,
                                          "flexGrow": 1,
                                          "flexShrink": 1,
                                        },
                                      ]
                                    }
                                  >
                                    <Text
                                      color="gray.1"
                                      fontFamily="default"
                                      fontSize={16}
                                      fontWeight="bold"
                                      lineHeight={22}
                                      numberOfLines={1}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#646464",
                                            "fontFamily": "Arial",
                                            "fontSize": 16,
                                            "fontWeight": "700",
                                            "lineHeight": 22,
                                          },
                                        ]
                                      }
                                    >
                                      Dr abc
                                    </Text>
                                    <Text
                                      color="gray.1"
                                      fontFamily="default"
                                      fontSize={16}
                                      lineHeight={22}
                                      numberOfLines={1}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#646464",
                                            "fontFamily": "Arial",
                                            "fontSize": 16,
                                            "lineHeight": 22,
                                          },
                                          Object {
                                            "textTransform": "capitalize",
                                          },
                                        ]
                                      }
                                    >
                                      General Practice
                                    </Text>
                                    <Text
                                      color="gray.1"
                                      fontFamily="default"
                                      fontSize={16}
                                      lineHeight={22}
                                      numberOfLines={1}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#646464",
                                            "fontFamily": "Arial",
                                            "fontSize": 16,
                                            "lineHeight": 22,
                                          },
                                        ]
                                      }
                                    >
                                      2223 2231
                                    </Text>
                                    <Text
                                      color="gray.1"
                                      fontFamily="default"
                                      fontSize={16}
                                      lineHeight={22}
                                      numberOfLines={2}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#646464",
                                            "fontFamily": "Arial",
                                            "fontSize": 16,
                                            "lineHeight": 22,
                                          },
                                        ]
                                      }
                                    >
                                      CommonWealth MRT
                                    </Text>
                                  </View>
                                  <View
                                    marginLeft={3}
                                    style={
                                      Array [
                                        Object {
                                          "marginLeft": 16,
                                        },
                                      ]
                                    }
                                  >
                                    <Image
                                      size={30}
                                      source={
                                        Object {
                                          "testUri": "../../@cxa-rn/components/src/ListItemWithRightChevron/images/angleRight.png",
                                        }
                                      }
                                      style={
                                        Array [
                                          Object {
                                            "height": 30,
                                            "width": 30,
                                          },
                                        ]
                                      }
                                    />
                                  </View>
                                </View>
                                <View
                                  pl={4}
                                  pr={4}
                                  style={
                                    Array [
                                      Object {
                                        "paddingLeft": 32,
                                        "paddingRight": 32,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    backgroundColor="gray.5"
                                    height={1}
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "#e3e3e3",
                                          "height": 1,
                                          "width": "100%",
                                        },
                                      ]
                                    }
                                    width="100%"
                                  />
                                </View>
                              </View>
                            </View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                accessibilityLabel="Dr xyz"
                                accessible={true}
                                focusable={true}
                                isTVSelectable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={null}
                              >
                                <View
                                  alignItems="center"
                                  flexDirection="row"
                                  px={4}
                                  py={24}
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "paddingBottom": 24,
                                        "paddingLeft": 32,
                                        "paddingRight": 32,
                                        "paddingTop": 24,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    flex={1}
                                    style={
                                      Array [
                                        Object {
                                          "flexBasis": 0,
                                          "flexGrow": 1,
                                          "flexShrink": 1,
                                        },
                                      ]
                                    }
                                  >
                                    <Text
                                      color="gray.1"
                                      fontFamily="default"
                                      fontSize={16}
                                      fontWeight="bold"
                                      lineHeight={22}
                                      numberOfLines={1}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#646464",
                                            "fontFamily": "Arial",
                                            "fontSize": 16,
                                            "fontWeight": "700",
                                            "lineHeight": 22,
                                          },
                                        ]
                                      }
                                    >
                                      Dr xyz
                                    </Text>
                                    <Text
                                      color="gray.1"
                                      fontFamily="default"
                                      fontSize={16}
                                      lineHeight={22}
                                      numberOfLines={1}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#646464",
                                            "fontFamily": "Arial",
                                            "fontSize": 16,
                                            "lineHeight": 22,
                                          },
                                          Object {
                                            "textTransform": "capitalize",
                                          },
                                        ]
                                      }
                                    >
                                      General Surgery
                                    </Text>
                                    <Text
                                      color="gray.1"
                                      fontFamily="default"
                                      fontSize={16}
                                      lineHeight={22}
                                      numberOfLines={1}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#646464",
                                            "fontFamily": "Arial",
                                            "fontSize": 16,
                                            "lineHeight": 22,
                                          },
                                        ]
                                      }
                                    >
                                      2223 2232, 2223 2233, 2223 2234
                                    </Text>
                                    <Text
                                      color="gray.1"
                                      fontFamily="default"
                                      fontSize={16}
                                      lineHeight={22}
                                      numberOfLines={2}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#646464",
                                            "fontFamily": "Arial",
                                            "fontSize": 16,
                                            "lineHeight": 22,
                                          },
                                        ]
                                      }
                                    >
                                      City Hall MRT
                                    </Text>
                                  </View>
                                  <View
                                    marginLeft={3}
                                    style={
                                      Array [
                                        Object {
                                          "marginLeft": 16,
                                        },
                                      ]
                                    }
                                  >
                                    <Image
                                      size={30}
                                      source={
                                        Object {
                                          "testUri": "../../@cxa-rn/components/src/ListItemWithRightChevron/images/angleRight.png",
                                        }
                                      }
                                      style={
                                        Array [
                                          Object {
                                            "height": 30,
                                            "width": 30,
                                          },
                                        ]
                                      }
                                    />
                                  </View>
                                </View>
                                <View
                                  pl={4}
                                  pr={4}
                                  style={
                                    Array [
                                      Object {
                                        "paddingLeft": 32,
                                        "paddingRight": 32,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    backgroundColor="gray.5"
                                    height={1}
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "#e3e3e3",
                                          "height": 1,
                                          "width": "100%",
                                        },
                                      ]
                                    }
                                    width="100%"
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
